equinix/config.go

```diff
@@ -33,18 +33,20 @@ const (
 // Config is the configuration structure used to instantiate the Equinix
 // provider.
 type Config struct {
-	BaseURL        string
-	AuthToken      string
-	ClientID       string
-	ClientSecret   string
-	MaxRetries     int
-	MaxRetryWait   time.Duration
-	RequestTimeout time.Duration
-	PageSize       int
-	Token          string
-
-	ecx              ecx.Client
-	ne               ne.Client
+	BaseURL        	string
+	AuthToken      	string
+	ClientID       	string
+	ClientSecret   	string
+	MaxRetries     	int
+	MaxRetryWait   	time.Duration
+	RequestTimeout	time.Duration
+	PageSize       	int
+	Token          	string
+
+	ecx		ecx.Client
+	ne		ne.Client
+	metal	*packngo.Client
```

Comment:
Moving the client into the Config should help the retryable HTTP transport/client to share error/retry timeouts and state. I think this will be a good thing.